pkgs/servers/web-apps/photoprism/default.nix

```diff
@@ -0,0 +1,127 @@
+{ lib
+, pkgs
+, stdenv
+, buildGoModule
+, fetchzip
+, fetchFromGitHub
+, coreutils
+, darktable
+, rawtherapee
+, ffmpeg
+, libheif
+, exiftool
+, nodejs
+, libtensorflow-bin
+}:
+
+let
+  version = "2020-09-21";
+  pname = "photoprism";
+
+  src = fetchFromGitHub {
+    owner = "photoprism";
+    repo = pname;
+    rev = "d030a602d47cf7279dddc301d74156ca9c21e7ef";
+    sha256 = "0gmvl4svcvdqb0y7chsn3dd13iznq4v3ms1rfvqcbrkhpqn3qhpd";
+  };
+
+  fetchModel = {name, sha256}:
+  fetchzip {
+    inherit sha256;
+    url = "https://dl.photoprism.org/tensorflow/${name}.zip";
+    stripRoot = false;
+  };
+
+  nasnet = fetchModel {
```

Comment:
This fetches the pretrained tensorflow models for classifying images.
The models don't have a stable URL, so I'm not sure how to best handle that. We could upload the models to archive.org.
In addition the models are quite large (~40MB and ~90MB), so I'm not sure if we should include them in the derivation at all.

Reply:
The size of the models doesn't bother me particularly.  You could also add an optional argument to the derivation to enable/disable them as well.  But you're right, we'd need a stable URL